x-pack/plugins/task_manager/server/lib/log_health_metrics.ts

```diff
@@ -25,23 +37,54 @@ export function logHealthMetrics(
     },
   };
   const statusWithoutCapacity = calculateHealthStatus(healthWithoutCapacity, config);
-  let logAsWarn = statusWithoutCapacity === HealthStatus.Warning;
-  const logAsError =
-    statusWithoutCapacity === HealthStatus.Error && !isEmpty(monitoredHealth.stats);
-  const driftInSeconds = (monitoredHealth.stats.runtime?.value.drift.p99 ?? 0) / 1000;
-
-  if (driftInSeconds >= config.monitored_stats_warn_delayed_task_start_in_seconds) {
-    logger.warn(
-      `Detected delay task start of ${driftInSeconds}s (which exceeds configured value of ${config.monitored_stats_warn_delayed_task_start_in_seconds}s)`
-    );
-    logAsWarn = true;
+  if (statusWithoutCapacity === HealthStatus.Warning) {
+    logLevel = LogLevel.Warn;
+  } else if (statusWithoutCapacity === HealthStatus.Error && !isEmpty(monitoredHealth.stats)) {
+    logLevel = LogLevel.Error;
   }
 
-  if (logAsError) {
-    logger.error(`Latest Monitored Stats: ${JSON.stringify(monitoredHealth)}`);
-  } else if (logAsWarn) {
-    logger.warn(`Latest Monitored Stats: ${JSON.stringify(monitoredHealth)}`);
+  const message = `Latest Monitored Stats: ${JSON.stringify(monitoredHealth)}`;
+  if (enabled) {
+    const driftInSeconds = (monitoredHealth.stats.runtime?.value.drift.p99 ?? 0) / 1000;
+    if (
+      driftInSeconds >= config.monitored_stats_health_verbose_log.warn_delayed_task_start_in_seconds
+    ) {
+      const taskType = Object.keys(monitoredHealth.stats.runtime?.value.drift_by_type ?? {}).reduce(
+        (accum: string, typeName) => {
+          if (
+            monitoredHealth.stats.runtime?.value.drift_by_type[typeName].p99 ===
```

Comment:
Could there be multiple task types with a drift exceeding the threshold? Maybe we should capture all of them?

Reply:
Good idea! I'll add support for that